src/patternfly/components/Button/button.scss

```
@@ -274,6 +281,10 @@
       --pf-c-button--m-primary--Color: var(--pf-c-button--m-primary--active--Color);
       --pf-c-button--m-primary--BackgroundColor: var(--pf-c-button--m-primary--active--BackgroundColor);
     }
+
+    &:not(:disabled):not(.pf-m-disabled) .pf-c-badge.pf-m-unread {
```

it looks like you might be able to drop the specificity here, if you just set `--pf-c-button--m-primary__c-badge--BorderWidth: 0` below in your `[disabled selectors] .pf-c-badge {}` block

```suggestion
    .pf-c-badge.pf-m-unread {
```

---

@mmenestr says it should only be used on the link button, however, if someone did put it on another button there were some bad things happening with contrast. So a minimal amount of css puts a guardrail there. I added a note in the documentation saying only to use it with link buttons.